### PR TITLE
Update to 0.4.x 

### PR DESCRIPTION
**Pull request by hfiguiere**
I was following the tutorial and it didn't work until I realised that it was still on 0.2.

This PR:

1. fix a small bug in one of the examples
2. Update both the examples and the documentation to the 0.4 API

I'm sure some corrections are missing, but this is IMHO better.

Let me know if you need me change anything else.

This should also supercede #1 in some way.

### CODE REVIEW

File: src/tracker.md [hunk 1]
```diff
@@ -17,16 +17,16 @@ When pressing a button, the icon above it will change. The background of the app
...
-+ `get_mut_#field_name()`  
++ `get_mut_#field_name()`
   Get a mutable reference to your field. Assumes the field will be modified and marks it as changed.
 
-+ `set_#field_name(value)`  
++ `set_#field_name(value)`
   Get a mutable reference to your field. Marks the field as changed only if the new value isn't equal with the previous value.
 
-+ `update_#field_name(fn)`  
++ `update_#field_name(fn)`
```
Comment (AaronErhardt) [inline src/tracker.md:29]: I think the spaces at the end of those lines were there on purpose to add a line break.

**Review by AaronErhardt — CHANGES_REQUESTED** [review]
Great work! This has been quite overdue, I'm really happy someone stepped up to do this :)

There's only one minor thing that should be addressed before merging.

File: src/tracker.md (hunk 1 above)
Comment (hfiguiere) [reply src/tracker.md]: reverted.

**Review by AaronErhardt — APPROVED** [review]
Thanks!

**Pull request merged**